content/docs/use-cases/kubeflow-on-multinode-cluster.md

```diff
@@ -0,0 +1,251 @@
+
+### Install Helm
+
+Helm is the Kubernetes packet manager. You can follow [this guide](https://helm.sh/docs/using_helm/#install-helm) to install it in a couple of steps.
+
+### Install NFS Provisioner
+
+The Dynamic Provisioner we are going to install will be able to create PVs for any PVC for a given [storage class](https://kubernetes.io/docs/concepts/storage/storage-classes/).
+
+A storage class is a label associated to volumes to specify a class of storage: using storage class 
+definition is possible to query and provision volumes with different performances or capabilities
+ (e.g SSD or slower disks)
+
+A special storage class is *default*: any PV or PVC that doesn't
+specify one is associated to it.
+
+NFS client provisioner can be found [here](https://github.com/helm/charts/tree/master/stable/nfs-client-provisioner).
+
+You can install it with Helm:
+
+```shell
+helm install 
+  --name nfs-client-provisioner \
+  --set nfs.server=<NFS Server IP> \
+  --set nfs.path=/exported/path \
+  --set storageClass.name=nfs \
+  --set storageClass.defaultClass=true \
+  stable/nfs-client-provisioner
+```
+
+The component in the system will add a *nfs* Storage Class that you can see in this way:
+
+```shell
+kubectl get storageclass -n kubeflow
+
+NAME                   PROVISIONER                            AGE
+nfs (default)          cluster.local/nfs-client-provisioner   6h13m
+...
+```
+
+As you noticed we specified an optional parameter storageClass.defaultClass to true: 
+this will define *nfs* storage class as default.
+In this way during KubeFlow installation, all PVCs will be labelled
+with *nfs* storage class.
+
+## Finally: install KubeFlow!
+
+We finally prepared our on-prem multi-node Kubernetes cluster to manage transparently volumes
+using a NFS server: now you can finally install KubeFlow following the Vanilla on-prem KubeFlow installation guideline that you encountered in the top of this document.
+
+At the end of procedure you'll see that all PVCs defined by KubeFlow will be binded to some
+automatically created PV with the storage class *nfs*.
+
+Notice that the provisioner will create a directory for each PVC with a unique name related 
+to that PVC inside the root NFS directory.
+
+## In case of existing kubeflow installation
+
+In case you setup Dynamic Provisioning *after* the installation of KubeFlow, you must change the storage class on your existing unbinded PVCs.
+
+Unlikely, Kubernetes doesn't allow to change storage class of PVCs on the fly.
+
+To perform this task you need to:
+
+* download existing PVCs.
+* change in them storage class
+* delete and recreate them in the cluster.
+
+Download the three PVCs:
+
+```shell
+kubectl get pvc/mysql-pv-claim -n kubeflow -o yaml > mysql-pv-claim.yaml
+kubectl get pvc/minio-pvc -n kubeflow -o yaml > minio-pvc.yaml
+kubectl get pvc/katib-mysql -n kubeflow -o yaml > katib.yaml
+```
+
+And then modify files to add the right storageClassName under *spec* section:
+
+```yaml
```

Comment:
Could you add a comment line to the start of each yaml snippet to clarify which file? For example, for this one:

```yaml
# mysql-pv-claim.yaml
apiVersion: v1
kind: PersistentVolumeClaim
...
```

(Do the same for the next two)

Reply:
I noticed also other bugs that I fixed: minio pvc was with a bad name and the storageClass was nfs-client instead nfs